ir/ir.def

```diff
@@ -154,7 +157,7 @@ class P4Control : Type_Declaration, INestedNamespace, ISimpleNamespace, IApply,
 
 /// A P4-16 action
 class P4Action : Declaration, ISimpleNamespace, IAnnotated, IFunctional {
-    optional Annotations        annotations = Annotations::empty;
```

Comment:
Would we get most/all of the memory savings by just using `inline` here instead of getting rid of IR::Annotations completely?

Reply:
Not all, as `IR::Annotations` will be still be a `Node` and it will still be cloned. Though "invisible" now when outer node is cloned by itself. Plus, there was already a FIXME for this change.

Reply:
> I think additional wrapper is not required as annotation is a property of the Node itself, so why do we need to have additional level of indirection?

I agree.